packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -953,7 +953,7 @@ export function createElement(type, config, children) {
 }
 
 export function cloneAndReplaceKey(oldElement, newKey) {
-  const clonedElement = ReactElement(
+  return ReactElement(
```

Comment:
This is a revert of https://github.com/facebook/react/pull/29662, which is no longer necessary.